### PR TITLE
Allows importing csv with coordinates

### LINKED ISSUE #332: Importer CSV avec coordonnées
Actuellement mviewer permet d'importer des fichiers CSV qui contiennent une adresse et de les afficher sur la carte grâce à un geocoding distant.

Il serait intéressant d'éteindre l'outil existant de permettre l'import des fichiers CSV qui déjà contiennent des coordonnées (n'importe le SCR/projection). Pour ce cas une configuration des projections supportées par l'outil me parait utile.

### PR DESCRIPTION
**Pull request by tkohr**
PR intents to resolve #332 and allows to import CSV files with coordinates. 

Projections that should be supported by the utility can be listed in XML config via proj4js strings.

### CODE REVIEW

File: js/csv.js [hunk 1]
```diff
@@ -160,8 +221,20 @@ var csv = (function () {
...
-                    //geocode file
-                    _geocode(evt.target.result,oLayer, oLayer.layer);
+                    
+                    if ($("#collapseZero").hasClass("in") !== false) {
+                        if ($("#srs-select").val()) {
+                            oLayer.xfield = $("#x-select").val();
+                            oLayer.yfield = $("#y-select").val();
+                            _mapCSV(evt.target.result, oLayer, oLayer.layer, $("#srs-select").val());
+                        } else {
+                            alert("Veuillez choisir une projection s'il vous plaît")
```
[the remark this answers is not in the document]
Comment (tkohr) [reply js/csv.js:231]: Indeed the entire csv tool is not internationalized so far. I could have a look at that.

File: demo/csv.xml [hunk 2]
```diff
@@ -47,6 +47,10 @@
                 yfield="latitude"
                 attribution="Dans cet exemple, l'utilisateur sélectionne un fichier csv depuis son poste. le fichier est ensuite géocodé après paramétrage du géocodage. Il n'y a pas de persistance du résultat. La recherche peut être activée en sélectionnant au moins un champ de recherche. exemple avec ce &lt;a href='https://raw.githubusercontent.com/geobretagne/mviewer/master/demo/demo.csv' target='_blank'>fichier &lt;/a>"
                 expanded="true">
+                <projections>
+                    <projection proj4js="'EPSG:3857','+proj=merc +a=6378137 +b=6378137 +lat_ts=0.0 +lon_0=0.0 +x_0=0.0 +y_0=0 +k=1.0 +units=m +nadgrids=@null +wktext  +no_defs'"/>
```
[the remark this answers is not in the document]
Comment (spelhate) [reply demo/csv.xml:51]: Pour info, c'est mentionné dans la doc : https://mviewerdoc.readthedocs.io/fr/latest/doc_tech/config_map.html?highlight=projextent

[the remark this answers is not in the document]
Comment (spelhate) [reply demo/csv.xml:51]: It is a new parameter. It's necessary to document it here: https://github.com/geobretagne/mviewer/blob/develop/docs/doc_tech/config_layers.rst

[the remark this answers is not in the document]
Comment (tkohr) [reply demo/csv.xml:51]: Thanks, I added the info to the doc

**Comment by Gaetanbrl** [comment]
Thanks for this PR ;) @tkohr !

* If you import CSV and clic on 'Choisir une projection' you could change fields and not just projection. So, first step is not really usefull.

Could we directly select `lat / long` with the second screen and pass the first ?

**Comment by Gaetanbrl** [comment]
* If you need to invert `lat `and `long`, you need to click on : 
1 - clic list lat
2 - select field
3 - cliclist long
4 - select field

It could be better to display a simple `revers `button between lat value and long value.
image

File: js/csv.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (tkohr) [reply js/csv.js]: Ok I've added i18n for templates and alert messages ;-)

**Comment by tkohr** [comment]
Thanks for the review @Gaetanbrl , good points.

I propose to keep the two "screens" to facilitate choosing x-y order and showing a small data extract at the same time, but indeed replace the selects with a reverse button.

**Comment by Gaetanbrl** [comment]
> I propose to keep the two "screens" to facilitate choosing x-y order

User don't know what is the correct order according he just select correct fields, no ?

**Comment by tkohr** [comment]
So finally I added a commit simplifying the UI to one step and adding a reverse button;-) Hope you like it @Gaetanbrl

**Comment by Gaetanbrl** [comment]
> Hope you like it @Gaetanbrl

Thanks.

Hope users like it ;)

**Comment by Gaetanbrl** [comment]
I really like it @tkohr ! nice work.

I have 2 little comments (very little) -> In commit line directly to target preview.

File: js/csv.js [hunk 3]
```diff
@@ -35,54 +35,40 @@ var csv = (function () {
...
+                                <th scope="col" i18n="csv.srs.projection">Projection (SRS)</th>
+                            </tr>
+                            </thead>
+                            <tbody>
+                            <tr>
+                                <th scope="row" i18n="csv.srs.select">Sélectionner</th>
+                                <td>
+                                    <select id="x-select" class="form-control"></select>
+                                </td>
+                                <td><button id="btn-xy" class="btn glyphicon glyphicon-refresh"></button></td>
```
Comment (Gaetanbrl) [inline js/csv.js:55]: Suggest to specify color like that btn-primary or btn-warning and size you like it :
`class="btn btn-sm btn-warning glyphicon glyphicon-refresh`

or 

`class="btn btn-sm btn-primary glyphicon glyphicon-refresh`

File: js/csv.js [hunk 4]
```diff
@@ -35,54 +35,40 @@ var csv = (function () {
...
+                                <td>
+                                    <select id="y-select" class="form-control"></select>
+                                </td>
+                                <td>
+                                    <select id="srs-select" class="form-control"></select>
+                                </td>
+                            </tr>
+                            <tr>
+                                <th scope="row" i18n="csv.srs.data">Extrait des données</th>
```
Comment (Gaetanbrl) [inline js/csv.js:64]: Could display `Aperçu ` (`en` : `Preview`) instead of "Extrait des données"

**Comment by spelhate** [comment]
> 2 suggestions but LGTM now.
> 
> Wait @spelhate returns ?

Yes Tomorow I think.

**Comment by spelhate** [comment]
Thank's @tkohr and @Gaetanbrl for review.

**Pull request merged**